File: 백준/Silver/10828. 스택/스택.c
#include <stdio.h>
#include <string.h>
int main() {
	int n, i, x;
	int stack[10005], top = 0;
	char s[6];

	scanf("%d", &n);

	for (i = 0; i < n; i++) {

		scanf("%s", &s);

		if (s[0] == 'p'&& s[1] == 'u') {
			scanf("%d", &x);
			top++;
			stack[top] = x;
		}

		else if (s[0] == 'p') {
			if (top < 1)
				printf("-1\n");
			else {
				printf("%d\n", stack[top]);
				top--;
			}
		}

		else if (s[0] == 's') {
			printf("%d\n", top);
		}

		else if (s[0] == 'e') {
			if (top < 1)
				printf("1\n");
			else
				printf("0\n");
		}

		else {
			if (top < 1)
				printf("-1\n");
			else {
				printf("%d\n", stack[top]);
			}
		}
	}
}

File: 백준/Silver/10828. 스택/스택.swift
import Foundation

var stack = [Int]()

for _ in 0..<Int(readLine()!)! {
    let line = readLine()!.split(separator: " ").map { String($0) }
    switch line[0] {
    case "push":
        stack.append(Int(line[1])!)
    case "pop":
        print(stack.isEmpty ? -1 : stack.removeLast())
    case "size":
        print(stack.count)
    case "empty":
        print(stack.isEmpty ? 1 : 0)
    case "top":
        print(stack.isEmpty ? -1 : stack[stack.count - 1])
    default: break
    }
}
